lib/gui/modules/drive-scanner.js

```diff
@@ -16,100 +16,150 @@
 
 'use strict';
 
-const Rx = require('rx');
 const os = require('os');
 const _ = require('lodash');
 const EventEmitter = require('events').EventEmitter;
 const drivelist = require('drivelist');
+const debug = require('debug')('etcher:drive-scanner');
 const settings = require('../models/settings');
 
-const DRIVE_SCANNER_INTERVAL_MS = 2000;
-const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 0;
-const emitter = new EventEmitter();
+const DRIVE_SCANNER_DELAY_MS = 2000;
+const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 125;
 
-/* eslint-disable lodash/prefer-lodash-method */
-
-const availableDrives = Rx.Observable.timer(
-  DRIVE_SCANNER_FIRST_SCAN_DELAY_MS,
-  DRIVE_SCANNER_INTERVAL_MS
-)
-
-/* eslint-enable lodash/prefer-lodash-method */
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ * @example
+ * var scanner = new DriveScanner()
+ *   .on('error', (error) => { ... })
+ *   .on('drives', (drives) => { ... })
+ */
+class DriveScanner extends EventEmitter {
```

Comment:
> Do we really need a whole class to perform this task? Just worrying that we're adding a lot of complexity (e.g. the number of code lines exploded compared to Rx) to perform a relatively simple task.

I don't see this being overly complex? Sure, that little that is going on here is visible now, but more than 50% of the lines are comments and whitespace, and the actual LOC inflation is relatively minimal, considering this uses no libs/frameworky things. Wrapping things in Promises will sure hide some of that little complexity and shave off a line or two, but I don't see that being worth it, or necessarily easier to reason about.

Plus, I think you'll end up with roughly the same LOC if doing it without a class, as you'll still have set up the state keeping vars, start(), stop() and the event emitter. Best to inherit from the EventEmitter, and keep all the context in one place, imo.

Does that make sense?